plugins/module_utils/scaleway.py

```diff
@@ -63,6 +73,101 @@ def parse_pagination_link(header):
         return parsed_relations
 
 
+def fetch_state(api, resource):
```

Comment:
This and other functions below in this file, they all seem to be receiving this `api` parameter which, if I got it right, it is an object of the class `Scaleway` also defined in this file.

Wouldn't it make more sense to add these functionalities as methods to that class, rather than external functions?

Reply:
You are right. I have implemented your suggestion in ac03fd5c5b30d99f508cb2699f9ff3513fabc44a